// @uiw/codemirror-theme-console/cjs/dark.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultSettingsConsoleDark = exports.consoleDarkInit = exports.consoleDark = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _codemirrorThemes = require("@uiw/codemirror-themes");
var defaultSettingsConsoleDark = exports.defaultSettingsConsoleDark = {
  background: '#000',
  foreground: '#fff',
  caret: '#fff',
  selection: '#5c6c9b',
  selectionMatch: '#2a3967',
  gutterBackground: '#000',
  gutterForeground: '#ada9a9',
  gutterActiveForeground: '#fff',
  lineHighlight: '#82828275'
};
var consoleDarkInit = exports.consoleDarkInit = function consoleDarkInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, defaultSettingsConsoleDark), settings),
    styles: (0, _toConsumableArray2["default"])(styles)
  });
};
var consoleDark = exports.consoleDark = consoleDarkInit();
