// @uiw/codemirror-theme-console/cjs/light.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultSettingsConsoleLight = exports.consoleLightInit = exports.consoleLight = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _codemirrorThemes = require("@uiw/codemirror-themes");
var defaultSettingsConsoleLight = exports.defaultSettingsConsoleLight = {
  background: '#fff',
  foreground: '#000',
  caret: '#000',
  selection: '#c2d2ff',
  selectionMatch: '#b1c1ef',
  gutterBackground: '#fff',
  gutterForeground: '#ada9a9',
  gutterActiveForeground: '#000',
  lineHighlight: '#c7c5c575'
};
var consoleLightInit = exports.consoleLightInit = function consoleLightInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, defaultSettingsConsoleLight), settings),
    styles: (0, _toConsumableArray2["default"])(styles)
  });
};
var consoleLight = exports.consoleLight = consoleLightInit();
